### DISCUSSION TITLE
Is it possible to run this on AMD GPUs? Also an output file question

### DISCUSSION THREAD

**Discussion by DL302** [nomadkaraoke/python-audio-separator#95] [Q&A] [answered]
Hello,
First I wanted to say how great this project is. I'm not very familiar with Python or machine learning, but this library is simple and isolates instrumentals quite well.

I have been running my Python code on a laptop with a discrete AMD GPU (on Linux, if that matters) and noticed that the library doesn't use the GPU at all for separating the instrumentals. I've tried using the `gpu` version of the library but it also doesn't use the GPU. Furthermore, I noticed in the README that there is no mention of using an AMD GPU. Are there steps I can take to get this library working to use my GPU? Or is this something that needs new code to support, either in this library or in the ONNXruntime library (which I noticed doesn't seem to have an AMD GPU version)?

Also, is it possible to specify a name for the output file? I have only seen an option to specify the output directory and I don't want to type `tmp/original_(Instrumental)_model_mel_band_roformer_ep_3005_sdr_11.wav` every time I want to use the instrumental.

**Comment by beveradb** [comment, marked as answer]
Hey @DL302 , sorry for the slow reply, I had a super busy week last week.
Glad it's helping you :)

Re. AMD GPUs - I wouldn't expect it to utilise them as I've never implemented any code to make that work (there was also no demand till now). Also FYI I don't even own an Nvidia GPU, never mind an AMD one 😅 so testing is always pretty inconvenient with me needing to spin up a Runpod container or VM somewhere to test GPU acceleration.
If you want to do some research into PyTorch GPU acceleration on AMD GPUs and raise a PR adding support for that, I would absolutely be happy to merge it as long as it looked sensible and I was able to test on a temporary AMD Runpod: https://www.runpod.io/pricing

No, it isn't possible to specify the name of the output file but it's only like 2 lines of code to rename it yourself 😅 
See https://github.com/nomadkaraoke/python-audio-separator/issues/89#issuecomment-2234119906
Again, PRs welcome if you want to implement that functionality!

**Comment by susman** [comment]
Tested demucs models with pytorch 2.4.1 and rocm 6.1.2 - seems to work fine.
